src/charm.py

```diff
@@ -298,27 +309,25 @@ def _deploy_k8s_resources(self) -> None:
             self.k8s_resource_handler.apply()
             self.crd_resource_handler.apply()
             self.configmap_resource_handler.apply()
-        except ApiError:
-            raise ErrorWithStatus("K8S resources creation failed", BlockedStatus)
+        except ApiError as e:
+            raise GenericCharmRuntimeError("Failed to create K8S resources") from e
         self.model.unit.status = MaintenanceStatus("K8S resources created")
 
     def _on_install(self, _):
         """Installation only tasks."""
         # deploy K8S resources to speed up deployment
         self._deploy_k8s_resources()
 
-    def _on_pebble_ready(self, _):
+    def _on_pebble_ready(self, event):
         """Configure started container."""
-        if not self.container.can_connect():
-            # Pebble Ready event should indicate that container is available
-            raise ErrorWithStatus("Pebble is ready and container is not ready", BlockedStatus)
+        self._check_container_connection(self.container)
```

Comment:
should we try except it?

Reply:
yes, as per our discussions in related PRs. Will update.

Reply:
@beliaev-maksim Updated, let me know what you think.

Reply:
@DnPlas I think we still put the charm into error state since we raise an error

Reply:
May be I misunderstood, but in this case charm should not be put in error state. Pebble not ready is not an error.

Reply:
@i-chvets correct, and we should not. Thus, need try/except

Reply:
LGTM

Reply:
no

Reply:
I see what you mean, will update.